plugin/src/main/java/org/opensearch/ml/action/handler/MLSearchHandler.java

```diff
@@ -140,20 +140,9 @@ public void search(SdkClient sdkClient, SearchRequest request, String tenantId,
                     .searchSourceBuilder(request.source())
                     .tenantId(tenantId)
                     .build();
-                sdkClient.searchDataObjectAsync(searchDataObjectRequest).whenComplete((r, throwable) -> {
-                    if (throwable == null) {
-                        try {
-                            SearchResponse searchResponse = SearchResponse.fromXContent(r.parser());
-                            log.info("Model search complete: {}", searchResponse.getHits().getTotalHits());
-                            doubleWrapperListener.onResponse(searchResponse);
-                        } catch (Exception e) {
-                            doubleWrapperListener.onFailure(e);
-                        }
-                    } else {
-                        Exception e = SdkClientUtils.unwrapAndConvertToException(throwable, OpenSearchStatusException.class);
-                        doubleWrapperListener.onFailure(e);
-                    }
-                });
+                sdkClient
+                    .searchDataObjectAsync(searchDataObjectRequest)
+                    .whenComplete(SdkClientUtils.wrapSearchCompletion(doubleWrapperListener));
```

Comment:
I think we should add some log [here](https://github.com/opensearch-project/opensearch-remote-metadata-sdk/blob/5c7d0053506a7ff3b9e1e8d5e1fa28be5f44f750/core/src/main/java/org/opensearch/remote/metadata/common/SdkClientUtils.java#L295-L311) to mimic the same experience?

Reply:
Sure, can do that.

Reply:
https://github.com/opensearch-project/opensearch-remote-metadata-sdk/pull/146